### PR TITLE
Add support for ConstBufferPointer on Vulkan.

### PR DESCRIPTION
**Pull request by csyonghe**
This change adds a `ConstBufferPointer` type that maps to a `buffer_reference` on Vulkan.

Usage:
```
struct MyStruct
{
    float4 position;
    float4x4 transform;
}

struct Globals
{
    ConstBufferPointer<MyStruct> pStruct;
}

ConstantBuffer<Globals> gGlobals;

RWStructuredBuffer<uint> outputBuffer;

[numthreads(1,1,1)]
void main(int3 tid: SV_DispatchThreadID)
{
    MyStruct s = gGlobals.pStruct.get();
    outputBuffer[tid.x] = uint(s.position.x);
}
```

### CODE REVIEW

File: source/slang/slang-ir.h [hunk 1]
```diff
@@ -1641,6 +1641,13 @@ struct IRRTTIPointerType : IRRawPointerTypeBase
     IR_LEAF_ISA(RTTIPointerType)
 };
 
+struct IRHLSLConstBufferPointerType : IRPtrTypeBase
+{
+    IR_LEAF_ISA(HLSLConstBufferPointerType)
+    IRInst* getElementType() { return getOperand(0); }
```
Comment (jsmall-nvidia) [inline source/slang/slang-ir.h:1647]: We already have `getValueType` on IRPtrTypeBase, is the use of element here to in some way distinguish this use case? (Say value might imply a single value, where element implies some amount of?)

Comment (csyonghe) [reply source/slang/slang-ir.h:1647]: Ah, you are right, I'll remove it.

File: source/slang/slang-ir.h [hunk 2]
```diff
@@ -1641,6 +1641,13 @@ struct IRRTTIPointerType : IRRawPointerTypeBase
...
+    IRInst* getAlignment() { return getOperand(1); }
+};
```
Comment (jsmall-nvidia) [inline source/slang/slang-ir.h:1649]: The alignment here is presumably the "element"/"value" alignment? Assuming that's right having it in the name might add to clarity.

Comment (jsmall-nvidia) [reply source/slang/slang-ir.h:1649]: Okay I see ```All buffer reference addresses used for a particular buffer
      reference type are assumed to be aligned to this alignment value. That
      is, the base of the block must be aligned, and members of the block can
      be aligned within the block using standard layouts and offset layout
      qualifiers.```

So perhaps it should have name like Base alignment or something like that.

**Review by jsmall-nvidia — DISMISSED** [review]

File: source/slang/slang-ir.h (hunk 2 above)
Comment (csyonghe) [reply source/slang/slang-ir.h:1648]: Renaming done.

File: source/slang/slang-ir.h (hunk 1 above)
Comment (csyonghe) [reply source/slang/slang-ir.h]: Fixed.

**Review by jsmall-nvidia — APPROVED** [review]

**Pull request merged**